### DISCUSSION TITLE
Proper usage for description-replace-character

### DISCUSSION THREAD

**Discussion by bethatasitmay** [PaloAltoNetworks/pan-os-php#679] [Q&A] [answered]
Hi - sorry, I forgot about my issue #407 and didn't notice it was fixed/closed.

I'm not getting the usage syntax correct.  The best I can get is...

*** SKIPPED : new and old description are the same

...and worst is some failure due to it just being wrong.

I have two iterations I'm trying, both with the potential for multiple words separated by spaces.

1) OldWord is at the beginning of the description
2) OldWord is near the beginning but always after a comma or semicolon ( ; )

Examples:

(1)

(A)
OldWord: Filtering Server
NewWord: nothing (leave description blank)

(B)
OldWord: Filtering Server, Load Balancer
NewWord: nothing (leave description blank)

(C - I don't think there are any of these)
OldWord: Filtering Server; Load Balancer
NewWord: nothing (leave description blank)

(2)

(A)
OldWord: ,Filtering Server
NewWord: Leave text preceding OldWord

(B)
OldWord: ,Filtering Server, Load Balancer
NewWord: Leave text preceding OldWord

(C)
OldWord: ,Filtering Server; Load Balancer
NewWord: Leave text preceding  OldWord

How would the action=description-replace-character be formed?  I understand the $$comma usage, but apparently that's it.

Again, if you want real example, I can email you.

Thanks!

**Comment by swaschkut** [comment, marked as answer]
(1)

(A)
OldWord: Filtering Server
NewWord: nothing (leave description blank)

`pan-os-php type=address 'actions=description-Replace-Character:Filtering Server'`

------------------------------------
2(A)
OldWord: ,Filtering Server
NewWord: Leave text preceding OldWord

`pan-os-php type=address 'actions=description-Replace-Character:$$comma$$Filtering Server'`


Based on these two example of course you need to focus on the order how you run these commands:

2A first then 1A, why:
if you run 1A first, then you only get ',' left

**Comment by bethatasitmay** [comment]
I tried that - here's the full output:

PowerShell version:

> php -r "require_once 'C:/tools/pan/pan-os-php/utils/pan-os-php.php';" type=address in=api://panorama.domain.com 'actions=description-Replace-Character:$$comma$$Filtering Server' 'filter=(name eq OBJECT_NAME)'

Translates to:

> pan-os-php type=address in=api://panorama.domain.com 'actions=description-Replace-Character:$$comma$$Filtering Server' 'filter=(name eq OBJECT_NAME)'

Output:

***********************************************
*********** pan-os-php.php type=address UTILITY **************

 - PAN-OS-PHP version: 2.0.62 [WIN] [8.1.10]
 - Downloading config from API...
 - Detected platform type is 'panorama'
 - No 'location' provided so using default ='shared'
 - No 'template' provided so using default ='any'
 - filter after sanitization : (name eq OBJECT_NAME)
 - Loading configuration through PAN-OS-PHP library...
   (1.97 seconds, 140.51 mb memory)
 - PAN-OS version: 91
 - PAN-OS APP-ID version: 8597-7479
 - PAN-OS Device timezone: US/Pacific is used. actual time: 2022/10/12 20:20:26

* processing store 'PanoramaConf: / AddressStore:addresses' that holds 7351 objects
   - object 'OBJECT_NAME' passing through Action='description-Replace-Character' Args: search=$$comma$$Filtering Server, replace=,
      *** SKIPPED : new and old description are the same

* objects processed in DG '' : 1

**** PROCESSING OF 1 OBJECTS DONE ****

************* END OF SCRIPT pan-os-php.php type=address ************

**Reply by swaschkut** [reply]
please mail me a full configuration file with these examples, where you are seeing the:
*** SKIPPED : new and old description are the same

by running your command

**Reply by swaschkut** [reply]
by the way:
as you are are not filtering this output:
*** SKIPPED : new and old description are the same

is available for every object which need to be processed.

--------------
   - object 'new' passing through Action='description-Replace-Character' Args: search=$$comma$$ TESTSERVER, replace=, 
      *** SKIPPED : new and old description are the same

   - object 'demo' passing through Action='description-Replace-Character' Args: search=$$comma$$ TESTSERVER, replace=, 
      * new description will be 'test ENDtest'
-----

**Comment by bethatasitmay** [comment]
By the way, what is the significance of the comma in "replace=," - is it just separating the output or is it indicating that the OldWord was to be replaced with a comma?

**Comment by bethatasitmay** [comment]
Oh, if it matters - ",Filtering Server" is one of many elements.  The elements (word or words separated by a comma and a space) fill the field (1,024 characters or less if there's something I want to keep before the first comma).

Let me know if you want me to email you some samples.

**Reply by swaschkut** [reply]
please send full configuration examples where you are seeing this issues by running PowerShell.
Please also the full command you where using.

**Comment by bethatasitmay** [comment]
My apologies - I made a mistake.  My filter was for the wrong rule name for the comma test.  I fixed that and it worked for the example above.

Is there a limit to how many words can be sent to it?  I'll email you shortly a couple of examples, but bottom line the descriptions are max length - 1024 characters.  The warning I see appears to be coming from PAN-OS-PHP or maybe PHP and not PS, but I could be wrong.

**Comment by bethatasitmay** [comment]
FYI, I just emailed you details.

**Reply by swaschkut** [reply]
what you need to do if a have a long description string:

`replace all , characters with $$comma$$`
